src/api/app/views/shared/_sourcediff.html.haml

```diff
@@ -0,0 +1,55 @@
+- editor_width ||= 'auto'
+- css_class ||= ''
+- css_id ||= nil
+- div_opts = { data: { url: url_for(controller: 'request', action: 'sourcediff') } }
+- div_opts[:id] = css_id
+- div_opts[:class] = "table_wrapper #{css_class}"
+- next_codemirror_uid
+- content_for(:content_for_head, javascript_include_tag('webui/application/cm2/index-diff'))
+= content_tag(:div, div_opts) do
+  - if filenames && filenames.length > 0
+    %table
+      %tbody
+        - last_file_id = nil
+        - filenames.each_with_index do |filename, index|
+          - file_element = files[filename]
+          - file_id = "#{valid_xml_id(filename)}_#{css_id}_#{index}"
+          - file_diff_text = file_element['diff'] || {}
+          - file_diff_text = file_diff_text['_content'] || ''
+          - file_state = file_element['state'].capitalize
+          - is_expanded = file_state != 'Deleted' && !filename.include?('/') && (filename == '_patchinfo' || filename.ends_with?('.spec') || filename.ends_with?('.changes'))
+          %tr{ id: "diff_headline_#{file_id}" }
+            %td{ style: 'width: 16px' }
+              - if file_diff_text.length > 0
+                %a{ href: '#', id: "collapse_#{file_id}", onclick: "collapse_expand(\'#{file_id}\'); return false;" } [-]
+                %a{ href: '#', id: "expand_#{file_id}", onclick: "collapse_expand(\'#{file_id}\'); return false;" } [+]
+                - content_for :ready_function do
+                  - if is_expanded
+                    $('#expand_#{file_id}').hide();
+                  - else
+                    $('#collapse_#{file_id}').hide();
+            %td{ style: 'width: 1%' }= file_state
+            %td
+              /
+                Most files don't exist when the package is linked, so keeping in mind the statement below don't even bother
```

Comment:
this comment can be shorted and written in a nicer way, removing for example the `...`